src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,165 @@
+package duke;
+
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.Storage.Storage;
+
+public class TaskList {
+
+    /** Initialising an empty array into which tasks can be added/manipulated/deleted */
+    private List<Task> tasks = new ArrayList<>();
+
+    /**
+     * This method provides a string that is the visual representation of the tasks seen by the user.
+     * @return The visual representation of task list.
+     */
+    public static String taskListString(List<Task> tasks) {
+        StringBuilder ans = new StringBuilder();
+        ans.append("Here are the tasks in your list:\n");
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            if (i != tasks.size() - 1) {
+                ans.append(String.format("%d. %s\n", i + 1, task.toString()));
+            } else {
+                ans.append(String.format("%d. %s", i + 1, task.toString()));
+            }
+        }
+        return ans.toString();
+    }
+
+    /**
+     * Initialise event subclass by type.
+     * @param task duke.Task string.
+     * @param type duke.Type enum.
+     * @param done Whether task is done or not Bool.
+     * @param localDateTime Datetime object.
+     * @return duke.Task of requisite type.
+     */
+    public static Task initialiseByType(String task, Type type, Boolean done, LocalDateTime localDateTime) {
+        if (type == Type.TODO) {
+            return new Todo(task, type, done);
+        } else if (type == Type.EVENT) {
+            return new Event(task, type, done, localDateTime);
+        } else {
+            return new Deadline(task, type, done, localDateTime);
+        }
+    }
+
+    /**
+     * The string that is outputted when user terminates bot.
+     * @return Message to say bye to user.
+     */
+    public String byeString() {
+        return "Bye. Hope to see you again soon!";
+    }
+
+    /**
+     * Assesses the input and activates the necessary response.
+     * @param input The string of input command.
+     * @throws DukeException Exceptions specific to this chatbot.
+     */
+    public String interpretInput(String input) throws DukeException {
```

Comment:
It could be better to extract out some of the processing logic into other methods to make this methed shorter and more readable.

Reply:
Thank you for the feedback Alex, will look to resolve this asap.